tslearn/metrics.py

```diff
@@ -643,26 +594,32 @@ def itakura_mask(sz1, sz2, max_slope=2.):
 
 
 def compute_mask(s1, s2, global_constraint=0,
-                 sakoe_chiba_radius=1, itakura_max_slope=2.):
+                 sakoe_chiba_radius=None, itakura_max_slope=None):
     """Compute the mask (region constraint).
 
     Parameters
     ----------
     s1 : array
         A time series.
+
     s2: array
         Another time series.
+
     global_constraint : {0, 1, 2} (default: 0)
```

Comment:
What is the reason for having this one in there as well? Is it for backward compatibility? Can't we just infer the type of constraint based on whether the values are set or not and remove this parameter?

Reply:
Backward compatibility implies that we have it for `dtw`, `dtw_path` & co. Then, we could remove this one but we would need to perform the checks in all these individual places, so I suggest we keep it here, unless you have a better idea of how to deal with it.

Reply:
Hmm don't we only need to do those checks in `compute_mask` and only there?

The checks would then be:

```python3
if sakoe_chiba_radius is not None and itakura_max_slope is not None:
  raise warning
elif sakoe_chiba_radius is not None:
  mask = sakoe_chiba_mask(sz1, sz2, radius=sakoe_chiba_radius)
elif itakura_max_slope is not None:
  mask = itakura_mask(sz1, sz2, max_slope=itakura_max_slope)
else:
  mask = numpy.zeros((sz1, sz2))
```

Reply:
Then, the case where `global_constraint` is set to either `sakoe_chiba` or `itakura` and both parameters are filled would end up raising a warning, which is a pity I would say.

Reply:
except if we remove it entirely or never use it. For backward compatibility, we could raise a warning that `global_constraint` is ignored/deprecated

Reply:
OK, but if we deprecate it, we should warn it in v0.3 and remove it actually in v0.5. I can add the deprecation warning if that seems ok to you.

Reply:
It is definitely not a necessity, it just seems a redundant parameter as we can infer it based on whether or not one of the variables is set to something else to None.